File: Gelman_BDA_ARM/arm/12.4.R
library("arm")
library("lme4")

srrs2 <- read.table ("../doc/gelman/ARM_Data/radon/srrs2.dat", header=T, sep=",")
mn <- srrs2$state=="MN"
radon <- srrs2$activity[mn]
log.radon <- log (ifelse (radon==0, .1, radon))
floor <- srrs2$floor[mn]       # 0 for basement, 1 for first floor
n <- length(radon)
y <- log.radon
x <- floor

# get county index variable
county.name <- as.vector(srrs2$county[mn])
uniq <- unique(county.name)
J <- length(uniq)
county <- rep (NA, J)

for (i in 1:J){
  county[county.name==uniq[i]] <- i
}

## Varying-intercept model w/ no predictors
M0 <- lmer (y ~ 1 + (1 | county))

coef(M0)
fixef (M0)
ranef (M0)
se.fixef(M0)
se.ranef(M0)

M1 <- lmer (y ~ x + (1 | county))
print (summary(M1))

coef(M1)
fixef (M1)
ranef (M1)
se.fixef(M1)
se.ranef(M1)

File: Gelman_BDA_ARM/arm/12.4.py
import rpy2.robjects as robjects
import rpy2.rinterface as rinterface
from rpy2.robjects.packages import importr
import rpy2.robjects.numpy2ri
import numpy as np

lme4 = importr("lme4")
srss2 = np.recfromcsv('../doc/gelman/ARM_Data/radon/srrs2.dat')
activity = srss2['activity']
state = srss2['state']
floor = srss2['floor']
county = srss2['county']

radon = activity[state == 'MN']
radon[radon == 0] = 0.1

floor = floor[state == 'MN']

log_radon = np.log(radon)

n = len(radon)
y = log_radon
x = floor

county_name = county[state == 'MN']
county = county[state == 'MN']
uniq = np.unique(county_name)
J = len(uniq)

cc = np.nan * np.ones(len(county))
for i in np.arange(J):
    cc[county_name==uniq[i]] = i+1

robjects.r.assign('y',y)
robjects.r.assign('x',x)
robjects.r.assign('county',county)

res = robjects.r.lmer('y ~ 1 + (1 | county)')
res = robjects.r.coef(res)
for x in res.iteritems(): 
    for i in range(len(x[1][0])):
        print "val", x[1][0][i]
